x-pack/plugins/security_solution/server/lib/detection_engine/signals/threat_mapping/get_threat_list.ts

```diff
@@ -52,18 +54,28 @@ export const getThreatList = async ({
       `Querying the indicator items from the index: "${index}" with searchAfter: "${searchAfter}" for up to ${calculatedPerPage} indicator items`
     )
   );
+
+  const conditionalConfig: {
+    _source?: string[] | boolean;
+    fields?: string[];
+  } = {};
+
+  if (threatMapping) {
+    conditionalConfig.fields = threatMapping
+      .map((mapping) => mapping.entries.map((item) => item.value))
+      .flat();
+    conditionalConfig._source = false;
+  } else {
+    conditionalConfig._source = [`${threatIndicatorPath}.*`, 'threat.feed.*'];
+  }
+
```

Comment:
Since these configs are set specially depending on where `getThreatList` is called from, it may be beneficial to allow `conditionalConfig` to be passed in to `getThreatList` as a parameter rather than computing it here. `createThreatList` basically owns the first conditional branch here and `buildThreatEnrichment` owns the second branch - changes to those two functions would require us to change the config they use in `getThreatList` as well.

If we move the config building logic into the functions that call `getThreatList` instead, we don't have to coordinate config changes between `getThreatList` and functions that call `getThreatList`. It also becomes clearer which config is used by each call to `getThreatList`.